src/crypto-providers/trezorCryptoProvider.ts

```diff
@@ -267,6 +313,23 @@ const TrezorCryptoProvider: () => Promise<CryptoProvider> = async () => {
     }
   }
 
+  const prepareTtl = (ttl?: number) => {
+    if (!ttl && !isFeatureSupportedForVersion(TrezorCryptoProviderFeature.ALLEGRA)) {
```

Comment:
I would not mix the concern of checking support of a feature and preparing the a piece of data for trezor - those things are not really related and the coupling brought in may just complicate testability and overall reasoning about the functions.

I'd rather centralize the feature checks in a single function to not clutter the "normal" code with such "petty" things, making it ultimately more readable. Here is an example of how I tackled it in Trezor connect:

https://github.com/trezor/connect/blob/65280a1a9d25a87d6cb55b95aa83faf0f5ec8d46/src/js/core/methods/CardanoSignTransaction.js#L132

Notice that I also generalized the function to return the error when given feature is not supported:

https://github.com/trezor/connect/blob/65280a1a9d25a87d6cb55b95aa83faf0f5ec8d46/src/js/core/methods/CardanoSignTransaction.js#L126

Similarly for Ledger

Reply:
I pulled out validations into one function. Creating generalization like in Trezor connect would be nice. We would need to refactor how our errors are done, right now they are defined strings and can't take arguments. I suggest we do that in another PR.